pkg/operator/apiserver/controllerset/apiservercontrollerset.go

```diff
@@ -276,6 +288,62 @@ func (cs *APIServerControllerSet) WithoutRevisionController() *APIServerControll
 	return cs
 }
 
+func (cs *APIServerControllerSet) WithEncryptionControllers(
+	component string,
+	encryptedGRs []schema.GroupResource,
+	migrationClient kubemigratorclient.Interface,
+	kubeClient kubernetes.Interface,
+	apiServerClient configv1client.APIServerInterface,
+	secretsClient corev1client.SecretsGetter,
+	apiServerInformer configv1informers.APIServerInformer,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	resourceSyncController resourcesynccontroller.ResourceSyncer,
+	deployer statemachine.Deployer,
```

Comment:
`deployer` is optional, a default one will be created if `nil`

for now, it will be different for `OAS-O` (`UnionRevisionLabelPodDeployer`) and `CAO`